pyanalyze/test_implementation.py

```diff
@@ -275,15 +274,15 @@ def capybara(s: str, b: bytes):
             assert_is_value(s.encode("utf-8"), TypedValue(bytes))
             assert_is_value(b.decode("utf-8"), TypedValue(str))
 
-    @assert_fails(ErrorCode.incompatible_argument)
+    @assert_passes()
     def test_encode_wrong_type(self):
         def capybara():
-            "".encode(42)
+            "".encode(42)  # E: incompatible_call  # E: incompatible_argument
```

Comment:
```suggestion
            # TODO this should produce only one error
            "".encode(42)  # E: incompatible_call  # E: incompatible_argument
```
Looks like one error is from the type checker, the other because we try to actually call the method. We should not attempt the call if type checking produced an error. But that's out of scope for this PR.